mteb/models/openai_models.py

```diff
@@ -15,27 +15,48 @@
 
 
 class OpenAIWrapper(Wrapper):
-    def __init__(self, model_name: str, embed_dim: int | None = None, **kwargs) -> None:
+    def __init__(
+        self,
+        model_name: str,
+        max_tokens: int,
+        tokenizer_name: str = "cl100k_base",  # since all models use this tokenizer now
+        embed_dim: int | None = None,
+        **kwargs,
+    ) -> None:
         requires_package(self, "openai", "Openai text embedding")
         from openai import OpenAI
 
         self._client = OpenAI()
         self._model_name = model_name
         self._embed_dim = embed_dim
+        self._max_tokens = max_tokens
+        self._tokenizer_name = tokenizer_name
 
     def encode(self, sentences: list[str], **kwargs: Any) -> np.ndarray:
         requires_package(self, "openai", "Openai text embedding")
+        requires_package(self, "tiktoken", "Tiktoken package")
+        import tiktoken
         from openai import NotGiven
 
         if self._model_name == "text-embedding-ada-002" and self._embed_dim is not None:
             logger.warning(
                 "Reducing embedding size available only for text-embedding-3-* models"
             )
 
+        trimmed_sentences = []
```

Comment:
Looks good. I would add a docstring to the model explaining this. 

I could e.g. imagine another alternative of embedding sequences and averaging the embedding and it is nice to be clear about what implementation choices we have made (this is in line with sentence transformers as far as I know).

Reply:
Do you mean adding a log explaining about trimming the sequence length ?

Reply:
No just a docstring for the class.

Something like:
"""
A wrapper around openAIs embedding API.

To handle documents larger than XXX we truncate the document to the specified sequence length.
"""

Reply:
Added.